engine/src/main/java/org/terasology/engine/persistence/internal/GlobalStoreLoader.java

```diff
@@ -112,7 +112,7 @@ private void loadComponentMapping(EntityData.GlobalStore globalStore) {
             if (componentMetadata != null) {
                 componentIdTable.put(componentMetadata.getType(), index);
             } else {
-                logger.warn("Unable to resolve component '{}'", globalStore.getComponentClass(index));
+                logger.atWarn().addArgument(globalStore.getComponentClass(index)).log("Unable to resolve component '{}'");
```

Comment:
`globalStore.getComponentClass(index)` is evaluated before the logger call
Please create and reference a local variable.